DSP Accel Library/Testing and Images/TestingLibraryScripts.m

```diff
@@ -0,0 +1,129 @@
+%% TESTING ONLY, USER DEFINED VARS
+clear;
+TableFn = "log(X_in)";
+maxInput = 1;
+floor_in_tab = 2^-12;
+error_cap_tab = 0.001;
+ERR_DIAG = true;
+N_bits_tab = 4;
+M_bits_tab = 5;
+igotthis = false;
+W_bits = 32;
+F_bits = 28;
+
+% things to display: table size, floor input?, max input, accuracy
+% things to figure out: M_bits, N_bits, Min_val, X_in table, Y_out table,
+% max_error, ram_size, table_size 
+
+d = ceil(log2(maxInput));
+if(igotthis)
+    N_bits = N_bits_tab;
+    M_bits = M_bits_tab;
+else
+    N_bits = ceil(log2(d-log2(floor_in_tab)+1));
+    % M_bits needs to be defined later, as it must be just enough to meet
+    % accuracy standards defined by user, with linear interpolation
+    M_bits = 1; %this will be updated later as needed
+end;
+
+repeatFlag = true;
+while(repeatFlag)
+    % Define a N point log2 spaced range of inputs, from 2^(d-(2^N_bits -1)) to almost 2^(d+1), 
+    % this effectively covers the user defined input range, down to a floor
+    % value.
+    X_in = zeros(1, 2^(M_bits+N_bits));
+    addr = 1;
+    for NShifts = 2^N_bits-1:-1:0
+        for M = 0:2^M_bits - 1
+            X_in(addr) = 2^(d-NShifts) + M*2^(d-NShifts-M_bits);
+            addr = addr+1;
+        end
+    end
+
+    % use function to define output. note: function must contain input X_in
+    % within the string to work properly, and the only output of the function
+    % must be Y_out.
+    Y_out = eval(TableFn);
+
+    ram_size = N_bits+M_bits;
+    Table_init = Y_out;
+    max_val = 2^d+ (2^M_bits -1)*2^(d-M_bits);
+    min_val = 2^(d-(2^N_bits-1));
```

Comment:
Same note as above for variable names.  Also, it would be nice to have a brief comment here to describe the variables.